src/main/kotlin/org/rust/lang/core/types/ty/TyStructOrEnum.kt

```diff
@@ -22,10 +24,23 @@ abstract class TyStructOrEnumBase(flags: TypeFlags) : Ty(flags) {
     abstract val item: RsStructOrEnumItemElement
 }
 
+private fun RsFieldsOwner.canBeCopyable(): Boolean {
+    return blockFields?.fieldDeclList?.all {
+        it.typeReference?.type?.isCopyable ?: false
+    } ?: tupleFields?.tupleFieldDeclList?.all {
+        it.typeReference.type.isCopyable
+    } ?: true
+}
+
 class TyStruct private constructor(
     private val boundElement: BoundElement<RsStructItem>
 ) : TyStructOrEnumBase(mergeFlags(boundElement)) {
 
+    override val isCopyable: Boolean get() {
+        val lookup = ImplLookup.relativeTo(boundElement.element)
+        return lookup.isCopy(this) && !lookup.isDrop(this) && boundElement.element.canBeCopyable()
```

Comment:
I think we should check only `Copy` trait implementation for this feature even rustc will emit an error while compiling.
These checks can be part of inspection to highlight incompatibilities in `Copy` implementation.